### PR TITLE
Update consistent hash ring docs with new index gateway ring

### PR DESCRIPTION
**Pull request by JordanRushing**
Signed-off-by: JordanRushing <rushing.jordan@gmail.com>

**What this PR does / why we need it**:

This PR updates the `Consistent Hash Ring` docs to include the newly-implemented IndexGateway ring (https://github.com/grafana/loki/pull/5358).

**Which issue(s) this PR fixes**:

N/A

**Special notes for your reviewer**:

N/A

**Checklist**
- [x] Documentation added
- [ ] Tests updated
- [ ] Is this an important fix or new feature? Add an entry in the `CHANGELOG.md`.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

**Review by DylanGuedes — APPROVED** [review]

**Review by kavirajk — APPROVED** [review]
LGMT :+1: 

Leave it to @KMiller-Grafana for final look and merge!

**Review by owen-d — APPROVED** [review]
LGTM

File: docs/sources/fundamentals/architecture/rings.md [hunk 1]
```diff
@@ -91,3 +94,7 @@ despite the compactor target being on multiple instances.
 ## About the ruler ring
 
 The ruler ring is used to determine which rulers evaluate which rule groups.
+
+## About the index gateway ring
+
+The index gateway ring is used to determine which gateway is responsible for which tenant's indices when queried by rulers or queriers.
```
Comment (KMiller-Grafana) [inline docs/sources/fundamentals/architecture/rings.md:100]: Indexes refers to more than one index.  Indices are for ranges, like an array index.
```suggestion
The index gateway ring is used to determine which gateway is responsible for which tenant's indexes when queried by rulers or queriers.
```

**Review by KMiller-Grafana — APPROVED** [review]
Please make the change I've suggested. I approve with or without it.

**Pull request merged**